clang/lib/Parse/ParseDecl.cpp

```diff
@@ -5055,6 +5058,17 @@ void Parser::ParseLexedCAttribute(LateParsedAttribute &LA, bool EnterScope,
   ParseGNUAttributeArgs(&LA.AttrName, LA.AttrNameLoc, Attrs, nullptr, nullptr,
                         SourceLocation(), ParsedAttr::Form::GNU(), nullptr);
 
+  const auto &SM = PP.getSourceManager();
+  CharSourceRange ExpansionRange = SM.getExpansionRange(LA.AttrNameLoc);
+  StringRef FoundName =
+      Lexer::getSourceText(ExpansionRange, SM, PP.getLangOpts())
+          .split('(')
+          .first;
+  IdentifierInfo *MacroII = PP.getIdentifierInfo(FoundName);
+  for (unsigned i = 0; i < Attrs.size(); ++i)
+    Attrs[i].setMacroIdentifier(MacroII, ExpansionRange.getBegin(),
```

Comment:
> when multiple are defined by the same one?

Do you have a specific example in mind? 

> Also, I find myself wondering if we should just use MacroIdentifier != nullptr instead of this? 

We could, I just wanted to make sure a system-defined macro is printed as macro, which likely provides a meaningful name, then user-defined macros. And doing so has much less impact on the existing regression tests.

Reply:
No specific example, but there are tons of macros that folks define to hold a giant collection of attributes (I think we have LLVM_INLINE or something like that which has a handful).  I just want to make sure that if we get that situation that we handle it reasonably.